packages/fuels-types/src/transaction.rs

```diff
@@ -0,0 +1,314 @@
+            vec![],
+            inputs.to_vec(),
+            outputs.to_vec(),
+            vec![],
+        )
+        .into()
+    }
+
+    /// Craft a transaction used to transfer funds to a contract.
+    pub fn build_contract_transfer_tx(
+        to: ContractId,
+        amount: u64,
+        asset_id: AssetId,
+        inputs: &[Input],
+        outputs: &[Output],
+        params: TxParameters,
+    ) -> ScriptTransaction {
+        let script_data: Vec<u8> = [
+            to.to_vec(),
+            amount.to_be_bytes().to_vec(),
+            asset_id.to_vec(),
+        ]
+        .into_iter()
+        .flatten()
+        .collect();
+
+        // This script loads:
+        //  - a pointer to the contract id,
+        //  - the actual amount
+        //  - a pointer to the asset id
+        // into the registers 0x10, 0x12, 0x13
+        // and calls the TR instruction
+        let script = vec![
+            op::gtf(0x10, 0x00, GTFArgs::ScriptData.into()),
+            op::addi(0x11, 0x10, ContractId::LEN as u16),
+            op::lw(0x12, 0x11, 0),
+            op::addi(0x13, 0x11, WORD_SIZE as u16),
+            op::tr(0x10, 0x12, 0x13),
+            op::ret(RegId::ONE),
+        ]
+        .into_iter()
+        .collect();
+
+        FuelTransaction::script(
+            params.gas_price,
+            params.gas_limit,
+            params.maturity,
+            script,
+            script_data,
+            inputs.to_vec(),
+            outputs.to_vec(),
+            vec![],
+        )
+        .into()
+    }
+
+    /// Craft a transaction used to transfer funds to the base chain.
+    pub fn build_message_to_output_tx(
+        to: Address,
+        amount: u64,
+        inputs: &[Input],
+        params: TxParameters,
+    ) -> ScriptTransaction {
+        let script_data: Vec<u8> = [to.to_vec(), amount.to_be_bytes().to_vec()]
+            .into_iter()
+            .flatten()
+            .collect();
+
+        // This script loads:
+        //  - a pointer to the recipient address,
+        //  - the amount
+        // into the registers 0x10, 0x11
+        // and calls the SMO instruction
+        let script = vec![
+            op::gtf(0x10, 0x00, GTFArgs::ScriptData.into()),
+            op::addi(0x11, 0x10, Bytes32::LEN as u16),
+            op::lw(0x11, 0x11, 0),
+            op::smo(0x10, 0x00, 0x00, 0x11),
+            op::ret(RegId::ONE),
+        ]
+        .into_iter()
+        .collect();
+
+        let outputs = vec![
+            // when signing a transaction, recipient and amount are set to zero
+            Output::message(Address::zeroed(), 0),
+            Output::change(to, 0, BASE_ASSET_ID),
+        ];
+
+        FuelTransaction::script(
+            params.gas_price,
+            params.gas_limit,
+            params.maturity,
+            script,
+            script_data,
+            inputs.to_vec(),
+            outputs.to_vec(),
+            vec![],
+        )
+        .into()
+    }
+}
```

Comment:
This is great, but I think the usage of a macro warrants at least 1 test to show that the implementation behaves the way we expect it to. This way, if we modify it, we can ensure we're not breaking anything.

Reply:
@iqdecay Are you talking about the `impl_tx_wrapper` macro? If so, in this case, if the macro implementation fails, it fails to comply with the implementation of the trait `Transaction`, failing right away at compile-time.

Reply:
I was thinking about the inside behavior, the same way we have unit tests for regular method implementation. Or are the methods here too simple?